ably/realtime_conn.go

```diff
@@ -116,6 +116,7 @@ func (c *Connection) dial(proto string, u *url.URL) (conn proto.Conn, err error)
 	query.Add("heartbeats", "true")
 	u.RawQuery = query.Encode()
 	timeout := c.opts.realtimeRequestTimeout()
+	// TODO: Use c.opts.baseCtx to dial.
```

Comment:
I disagree with this, if we need a context here, it should be passed through the call stack, or derived from the options (e.g. `context.WithTimeout(context.Background(), c.opts.DialTimeout)`)

Reply:
Comment removed at 3b04db76d797942ccfb586ab52e1b3fdcc9b37bb.